src/gwt/src/org/rstudio/studio/client/workbench/prefs/views/PaneLayoutPreferencesPane.java

```diff
@@ -320,13 +331,121 @@ else if (!otherIndices.get(i))
       updateTabSetLabels();
    }
 
+   private String updateTable(int newCount)
+   {
+      // nothing has changed since the last update
+      if (grid_ != null && displayColumnCount_ == newCount)
+         return "";
+
+      // cells will be twice a wide as columns to preserve space
+      double columnCount = newCount + (2 * GRID_PANE_COUNT);
+      double columnWidthValue = (double)TABLE_WIDTH / columnCount;
+      double cellWidthValue = columnWidthValue * GRID_PANE_COUNT;
+
+      // If the column width is bigger than MAX_COLUMN_WIDTH, give space back to the panes
+      if (newCount > 0 && Math.min(columnWidthValue, MAX_COLUMN_WIDTH) != columnWidthValue)
+      {
+         double extra = (newCount * (columnWidthValue - MAX_COLUMN_WIDTH)) / GRID_PANE_COUNT;
+         cellWidthValue += extra;
+         columnWidthValue = MAX_COLUMN_WIDTH;
+      }
+      cellWidthValue -= (GRID_CELL_SPACING + GRID_CELL_PADDING);
+      columnWidthValue -= (GRID_CELL_SPACING + GRID_CELL_PADDING);
+
+      final String columnWidth = columnWidthValue + "px";
+      final String cellWidth = cellWidthValue + "px";
+      leftTop_.setWidth(cellWidth);
+      leftBottom_.setWidth(cellWidth);
+      rightTop_.setWidth(cellWidth);
+      rightBottom_.setWidth(cellWidth);
+
+      // create grid
+      if (grid_ == null)
+      {
+         grid_ = new FlexTable();
+         grid_.addStyleName(res_.styles().paneLayoutTable());
+         grid_.setCellSpacing(GRID_CELL_SPACING);
+         grid_.setCellPadding(GRID_CELL_PADDING);
+
```

Comment:
Recommend giving this table an accessible name; this is a cheap but effective accessibility improvement for screen reader users.

         Roles.getGridRole().setAriaLabelProperty(grid_.getElement(), "Columns and Panes Layout");